pandas/tests/groupby/test_transform.py

```diff
@@ -1001,3 +1001,46 @@ def test_ffill_not_in_axis(func, key, val):
     expected = df
 
     assert_frame_equal(result, expected)
+
```

Comment:
so we need an additional test here. you are partitioning the whitelist into 3 things: reductions, transforms, and other. We need to assert that if another function is added to the whitelist it is *also* added to one of these three categories. So I would add where you define the partitioning (in pandas.core.groupby.base) to actually define the other   (as a list of functions); 

then you can have a test that asserts the above; automatically testing that we satisfy the invariant (this could also be done via asserts in the code), but i think a test is more informative.

Reply:
> We need to assert that if another function is added to the whitelist

Not if the whitelist is a strictly defined as the union of the two. Would an comment be enough? "NOTE: do not edit this directly!". What we want to enforce is that every future addition must be classified as one or the other.

Reply:
a comment is not enough 

need a test to enforce this

Reply:
> so we need an additional test here. you are partitioning the whitelist into 3 things: reductions, 
> transforms, and other.

I'm not. `transform_kernel_whitelist` is by definition the union of reductions and transforms. Other things don't belong in the whitelist.

Reply:
> you are partitioning the whitelist into 3 things: 

What is contained in the `other` category? take `corr` for example, which I initially included by mistake as a reduction. It returns multiple items per group, so its not a reduction. and it returns a multiIndex, so it isn't like-indexed which is the definition of a transformation.

I don't think it has any bearing on `transform(name)`, it's just an unrelated method on Grouper.
So I don't think there's any third category to have in the whitelist for `transform`. If you don't agree, can you give an example?

Reply:
other = whitelist - reductions - transforms. we have to cover *every* single method.

basically the leftovers. put it it this way, if you add a function to the grouper (or change something in the reductions / transforms / other), then this test should fail. you have to do *both*. this ensures that these are always updated in sync.

doc comments don't work here.

Reply:
ah, that makes sense. Will do.